core/network-libp2p/src/custom_proto/behaviour.rs

```diff
@@ -372,10 +372,33 @@ impl<TMessage, TSubstream> CustomProtos<TMessage, TSubstream> {
 				.count();
 			self.max_outgoing_connections - num_outgoing_connections
 		};
-
 		trace!(target: "sub-libp2p", "Connect-to-nodes round; attempting to fill {:?} slots",
 			num_to_open);
 
+		// We first try to enable existing connections.
+		for peer_id in &self.connected_peers {
+			if num_to_open == 0 {
```

Comment:
Shouldn't this if be outside at the outside of this for loop?

Reply:
Oh, that's because I forgot to add `num_to_open -= 1;` inside the loop.